src/test/java/teammates/logic/core/NotificationsLogicTest.java

```diff
@@ -55,14 +63,43 @@ private void testGetNotification() throws Exception {
         assertNull(notifLogic.getNotification("nonexistant-notification"));
     }
 
+    private void testGetAllNotifications() throws Exception {
+        ______TS("success: retrieve all when there exist notifications");
+
+        n = getTypicalNotification();
+        notifDb.createEntity(n);
+        NotificationAttributes n1 = getTypicalNotification();
+        n1.setNotificationId("new-notification-id");
+        notifDb.createEntity(n1);
+
+        List<NotificationAttributes> retreivedNotifs = notifLogic.getAllNotifications();
+        retreivedNotifs.forEach(this::verifyPresentInDatabase);
```

Comment:
Probably not necessary for this. However, you may want to check whether the returned notif are indeed the list containing `n` and `n1`.

Reply:
Since we have created entities of `n` and `n1` in the database, by `verifyPresentInDatabase`, we make sure the list includes all those in DB, so I think this method will be more complete compared to checking n and n1 only.

Reply:
I will improve on this by checking all notifications form typical data bundle are fetched instead of verifying whether they are in database.